go/adcm/status/handlers.go

```diff
@@ -295,10 +295,67 @@ func showHost(h Hub, w http.ResponseWriter, r *http.Request) {
 		ErrOut4(w, r, "HOST_NOT_FOUND", "unknown host")
 		return
 	}
-	val, _ := h.HostStorage.get(ALL, hostId)
+	val, _ := h.HostStatusStorage.get(ALL, hostId)
 	jsonOut(w, r, val)
 }
 
+func listHost(h Hub, w http.ResponseWriter, r *http.Request) {
+	allow(w, "GET, POST")
+	result, _ := h.ListHostStorage.list()
+	jsonOut(w, r, result)
+
+}
+
+func createHost(h Hub, w http.ResponseWriter, r *http.Request) {
+	allow(w, "GET, POST")
+	hosts := make([]Host, 0)
+	_, err := decodeBody(w, r, &hosts)
+	if err != nil {
+		ErrOut4(w, r, "JSON_ERROR", err.Error())
+		return
+	}
+	code := h.ListHostStorage.create(hosts)
+	logg.D.f("createHost: %+v", hosts)
+	jsonOut3(w, r, "", code)
+
+}
+
+func retrieveHost(h Hub, w http.ResponseWriter, r *http.Request) {
+	allow(w, "GET, PUT")
+	hostId, ok := getPathId(w, r, "hostid")
+	if !ok {
+		return
+	}
+	_, ok = h.ServiceMap.getHostCluster(hostId)
+	if !ok {
+		ErrOut4(w, r, "HOST_NOT_FOUND", "unknown host")
+		return
+	}
+	value, _ := h.HostStorage.retrieve(hostId)
+	jsonOut(w, r, value)
+}
+
+func updateHost(h Hub, w http.ResponseWriter, r *http.Request) {
```

Comment:
Maybe "updateHostInfo"? Because in statusserver we may update host status, component on host status, host's MM and maybe even host's presence in a cluster